From 7e7acd1320f1d98c3c917b360b6e6c8db6202e9f Mon Sep 17 00:00:00 2001
From: Kai Pastor <dg0yt@darc.de>
Date: Mon, 22 Dec 2014 08:54:10 +0100
Subject: [PATCH] Android: Don't assume native window decoration in QDockWidget

Android does not decorate a floating QDockWidget, leaving the user with
no option to close or move such a widget. This changes activates Qt's
own drawing of window decorations for this case (similar to Win CE).

[ChangeLog][Android][QtWidgets] Enable QDockWidget window decorations.

Change-Id: Id46a994a8be9b8f3e7b530af118315cd36b5fb0e
---
 src/widgets/widgets/qdockwidget.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/widgets/widgets/qdockwidget.cpp b/src/widgets/widgets/qdockwidget.cpp
index 121fb2d..01a1f50 100644
--- a/src/widgets/widgets/qdockwidget.cpp
+++ b/src/widgets/widgets/qdockwidget.cpp
@@ -207,7 +207,7 @@ static bool isXcb()
 
 bool QDockWidgetLayout::nativeWindowDeco(bool floating) const
 {
-#ifdef Q_OS_WINCE
+#if defined(Q_OS_WINCE) || defined(Q_OS_ANDROID)
     return false;
 #else
     return !isXcb() && (floating && item_list[QDockWidgetLayout::TitleBar] == 0);
-- 
1.9.1

